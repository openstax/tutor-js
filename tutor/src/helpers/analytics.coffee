_ = require 'underscore'

Routes = require '../routes'

{CurrentUserStore} = require '../flux/current-user'

DestinationHelper  = require '../helpers/routes-and-destinations'

# generate custom event data for routes
Events =
  viewTaskStep: ({courseId}) ->
    # track all work done on a course
    Analytics.sendEvent 'Course', 'Work', label: courseId

  viewStudentDashboard: ({courseId}) ->
    # compare activity between courses
    Analytics.sendEvent 'Student', 'Dashboard', label: courseId


# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
Translators =

  dashboard:            ({courseId}) -> "/student/choose-course/#{courseId}"
  viewPractice:         ({courseId}) -> "/student/practice/#{courseId}"
  viewPerformanceForecast:  ({courseId}) -> "/student/performance-forecast/#{courseId}"
  viewStudentDashboard: ({courseId}) -> "/student/dashboard/#{courseId}"

  calendarByDate:        ({courseId}) -> "/teacher/calendar/#{courseId}"
  viewTeacherPerformanceForecast:      ({courseId}) -> "/teacher/performance-forecast/#{courseId}"
  viewScores:            ({courseId}) -> "/teacher/student-scores/#{courseId}"
  courseSettings:        ({courseId}) -> "/teacher/roster/#{courseId}"
  editReading:           ({courseId}) -> "/teacher/assignment/edit/reading/#{courseId}"
  editHomework:          ({courseId}) -> "/teacher/assignment/edit/homework/#{courseId}"
  editExternal:          ({courseId}) -> "/teacher/assignment/edit/external/#{courseId}"
  createReading:         ({courseId}) -> "/teacher/assignment/create/reading/#{courseId}"
  createHomework:        ({courseId}) -> "/teacher/assignment/create/homework/#{courseId}"
  createExternal:        ({courseId}) -> "/teacher/assignment/create/external/#{courseId}"
  calendarViewPlanStats: ({courseId}) -> "/teacher/metrics/quick/#{courseId}"
  reviewTask:            ({courseId}) -> "/teacher/metrics/review/#{courseId}"

  viewReferenceBook:        ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookSection: ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookPage:    ({courseId}) -> "/reference-view/#{courseId}"

  # Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep:         ({courseId}) ->
    role = CurrentUserStore.getCourseRole(courseId, true)
    "/#{role}/task-step/#{courseId}"


GA = undefined

Analytics =

  setTracker: (tracker) -> GA = tracker

  sendPageView: (url) ->
    GA?('send', 'pageview', url)

  onNavigation: (path) ->
    return unless GA

    route = Routes.pathToEntry(path)

    return @sendPageView("/not-found/#{change.path}") unless route

    path = Translators[route.entry.name]?( route.match.params ) or route.match.pathname

    # if we're also going to send custom events then we set the page
    if Events[route.entry.name]
      GA('set', 'page', path)
      Events[route.entry.name]( route.match.params )
      @sendPageView() # url's not needed since it was set before events
    else
      @sendPageView(route.match.pathname)

  sendEvent: (category, action, attrs) ->
    return unless GA
    GA('send', 'event', category, action, attrs.label, attrs.value)

module.exports = Analytics
