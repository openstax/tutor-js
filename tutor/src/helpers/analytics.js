import { partial, get } from 'lodash';

import Router from '../helpers/router';
import Courses from '../models/courses-map';

// generate custom event data for routes
const Events = {
  viewTaskStep({ courseId }) {
    // track all work done on a course
    return Analytics.sendEvent('Course', 'Work', { label: courseId });
  },

  viewStudentDashboard({ courseId }) {
    // compare activity between courses
    return Analytics.sendEvent('Student', 'Dashboard', { label: courseId });
  },
};

// a bit shorter helper methods
const isTeacher = courseId => get(Courses.get(courseId), 'currentRole.isTeacher', false);
const getRole = function(courseId) {
  if (Courses.get(courseId).is_teacher) { return 'teacher'; } else { return 'student'; }
};

const assignmentTypeTranslator = function(assignmentType, { courseId, id }) {
  const newOrEdit = id === 'new' ? 'create' : 'edit';
  return `/teacher/assignment/${newOrEdit}/${assignmentType}/${courseId}`;
};

function viewReferenceBook({ courseId, pageId }) {
  const url = `/reference-view/${courseId}`;
  if (!pageId) {
    return url;
  }
  const course = Courses.get(courseId);
  const page = course.referenceBook.pages.byId.get(pageId);
  return page ? `${url}/section/${page.chapter_section.key}` : url;
}

// Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
const Translators = {
  dashboard({ courseId }) {
    if (isTeacher(courseId)) { return `/teacher/calendar/${courseId}`; } else { return `/student/dashboard/${courseId}`; }
  },
  practiceTopics({ courseId }) { return `/student/practice/${courseId}`; },
  viewPerformanceGuide({ courseId }) {
    return `/${getRole(courseId)}/performance-forecast/${courseId}`;
  },
  calendarByDate({ courseId }) { return `/teacher/calendar/${courseId}`; },
  viewGradebook({ courseId }) { return `/teacher/student-scores/${courseId}`; },
  courseSettings({ courseId }) { return `/teacher/roster/${courseId}`; },
  editAssignment: partial(assignmentTypeTranslator, 'assignment'),
  calendarViewPlanStats({ courseId }) { return `/teacher/metrics/quick/${courseId}`; },
  reviewTask({ courseId }) { return `/teacher/metrics/review/${courseId}`; },
  viewReferenceBookSection: viewReferenceBook,
  viewReferenceBookPage: viewReferenceBook,
  viewReferenceBook,

  // Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep({ courseId }) {
    const course = Courses.get(courseId);
    const role = course ? course.primaryRole.type : 'unknown';
    return `/${role}/task-step/${courseId}`;
  },
};

const TAG_MANAGER_ID = 'GTM-W6N7PB';

let GA = undefined;
let trackerNames = undefined;

var Analytics = {
  // copied from BIT
  initTagManager(w, d, s, l, i) {
    // Disable ESLint rules since we're copying Google's script
    /* eslint-disable one-var, prefer-const, prefer-template */
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    let f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l !== 'dataLayer' ? '&l=' + l : '';
    // Breaks in tests because there are no scripts
    if (f) {
      j.async = true;
      j.src = '//www.googletagmanager.com/gtm.js?id=' + i + dl;
      f.parentNode.insertBefore(j, f);
    }
  },
  
  // Set the Command Queue (ga) and cache the (possibly multiple) tracker names
  setGa(ga) {
    trackerNames = undefined;
    GA = ga;
    if (GA) {
      GA(function() {
        trackerNames = GA.getAll().map(function(tracker) { return tracker.get('name'); });
      });
      this.initTagManager(window, document, 'script', 'dataLayer', TAG_MANAGER_ID);
    }

    return GA;
  },

  // Calls all trackers with the command and arguments given here
  ga(command, ...params) {
    if (!GA || !trackerNames) { return; }
    trackerNames.forEach((trackerName) => {
      this.realGa(trackerName + '.' + command, ...params);
    });
  },

  // Exists largely for spec mocking
  realGa(...args) {
    GA(...args);
  },

  sendPageView(url) {
    this.ga('send', 'pageview', url);
  },

  recordCourseDimension(courseId) {
    const course = Courses.get(courseId);
    if (course) {
      this.ga('set', 'dimension1', course.appearance_code);
    }
  },

  onNavigation(path) {
    if (!GA) { return; }
    const route = Router.currentMatch(path);

    if (!route) {
      this.sendPageView(`/not-found${path}`);
      return;
    }
    const { courseId } = route.params;
    if (courseId) { this.recordCourseDimension(courseId); }
    const translatedPath = Translators[route.entry.name] ?
      Translators[route.entry.name]( route.params ) : route.pathname;

    this.ga('set', 'page', translatedPath);
    // if we're also going to send custom events then we set the page
    if (Events[route.entry.name]) {
      Events[route.entry.name]( route.params );
    }

    this.sendPageView(); // url's not needed since it was set before events
  },

  sendEvent(category, action, attrs = {}) {
    this.ga('send', 'event', category, action, attrs.label, attrs.value);
  },
};

export default Analytics;
