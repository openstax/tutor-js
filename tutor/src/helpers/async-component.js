import React from 'react';
import AsyncErrorBoundary from '../components/error-monitoring/async-load-error';
import LoadingAnimation from '../../../shared/src/components/loading-animation'

export function asyncComponent(loader, name = '') {

    const Component = React.lazy(loader);
    const loading = <LoadingAnimation message={`Loading ${name}â€¦`} />;

    const Loader = (props) => (
        <AsyncErrorBoundary>
            <React.Suspense fallback={loading}>
                <Component {...props} />
            </React.Suspense>
        </AsyncErrorBoundary>
    );

    return (props) => <Loader {...props} />;
}


export function loadAsync(resolve, name) {
    // return a function so the router will only evaluate it when it's needed
    // in the future we may insert role dependant logic here
    return () => asyncComponent(resolve, name);
}
