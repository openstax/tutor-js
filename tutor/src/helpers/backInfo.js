import { navHistory } from '../models'


const backInfo = () => {
    // Gets route to last path that was not the same as the current one
    // See TransitionStore for more detail.
    const historyInfo = navHistory.latest

    const text = historyInfo != null && !!historyInfo.name ? `Back to ${historyInfo.name}` : 'My Courses';
    const to =  historyInfo != null && !!historyInfo.path ? historyInfo.path : 'myCourses';

    return {
        text,
        to,
    };
};


export default backInfo;
