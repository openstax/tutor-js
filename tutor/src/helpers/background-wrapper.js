import { styled } from 'vendor';
import { navbars } from '../theme.js';

const BackgroundWrapper = styled.div`
  background: #fff;
  min-height: calc(100vh - ${navbars.top.height} - ${navbars.bottom.height});
  position: relative;
  overflow: hidden;
  padding: 0 2.4rem;
`;

const ContentWrapper = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

export { BackgroundWrapper, ContentWrapper };
