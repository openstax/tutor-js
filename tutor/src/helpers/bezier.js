function EASE_IN_OUT(t) {
    if (t < .5) { return 4 * t * t * t; } else { return ((t - 1) * ((2 * t) - 2) * ((2 * t) - 2)) + 1; }
}

function CALC(start, end, elapsed, duration) {
    if (elapsed > duration) { return end; }
    return start + ((end - start) * EASE_IN_OUT(elapsed / duration));
}

export default function bezier({
    range, duration, onStep, onComplete,
}) {
    const startTime = Date.now();
    const requestAnimationFrame = window.requestAnimationFrame || window.setTimeout;

    const step = () => {
        const elapsed = Date.now() - startTime;
        onStep(CALC(range[0], range[1], elapsed, duration));
        if (elapsed < duration) {
            requestAnimationFrame(step);
        } else if (onComplete) {
            onComplete();
        }
    };

    step();
}
