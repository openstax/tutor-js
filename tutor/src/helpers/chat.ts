import { Raven } from '../models'

const DEPLOYMENT_ID   = '5724P000000ghju';
const ORGANIZATION_ID = '00DU0000000Kwch';
const ELEMENT_ID      = '5736f000000k9eQ';


const win = window as any

export const Chat = {
    initialized: false,

    get isEnabled() {
        return Boolean(win.liveagent);
    },

    get isOnline() {
        if (!this.isEnabled) { return false; }
        const chat = document.querySelector<HTMLDivElement>('.chat.enabled');
        return chat && chat.style.display !== 'none';
    },

    initialize() {
    // will emit debugging info to the console
    // liveagent.enableLogging();
    },

    setElementVisiblity(online: any, offline?: any) {
        if (!this.isEnabled) { return; }
        try {
            win.liveagent.showWhenOnline(ELEMENT_ID, online);
            if (offline) {
                win.liveagent.showWhenOffline(ELEMENT_ID, offline);
            }
            if (!this.initialized) {
                win.liveagent.init('https://d.la1-c2-ia5.salesforceliveagent.com/chat', DEPLOYMENT_ID, ORGANIZATION_ID);
                this.initialized = true;
            }
        } catch(err) {
            Raven.captureException(err);
        }
    },

    start() {
        win.liveagent.startChat(ELEMENT_ID);
    },

};
