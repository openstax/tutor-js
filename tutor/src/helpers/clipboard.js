import _ from 'underscore';

const SUPPORTED = typeof document.queryCommandSupported === 'function' ? document.queryCommandSupported('copy') : undefined;

export default {
    isSupported: _.constant(!!SUPPORTED),

    copy() {
        if (!SUPPORTED) { return; }

        try {
            document.execCommand('copy');
        } catch (e) {
            // eslint-disable-next-line no-console
            console.warn('clipboard copy failed', e);
        }
    },
};
