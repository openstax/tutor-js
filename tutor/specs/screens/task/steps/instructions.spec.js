import { Factory, C } from '../../../helpers';
import Instructions from '../../../../src/screens/task/step/instructions';

describe('Instructions', () => {
    let task;
    let props;

    beforeEach(() => {
        task = Factory.studentTask({
            type: 'homework',
            stepCount: 1,
            due_at: '20210816T200000Z',
            closes_at: '20210823T200000Z',
        });
        const ux = { canGoForward: true, goForward: jest.fn(), goToStepId: jest.fn(), task: task };
        props = { ux };
    });

    it('renders the assignment instructions for immediate feedback', () => {
        task.auto_grading_feedback_on = 'answer';
        task.manual_grading_feedback_on = 'grade';
        expect.snapshot(<C><Instructions {...props} /></C>).toMatchSnapshot();
    });

    it('renders the assignment instructions for feedback when due', () => {
        task.auto_grading_feedback_on = 'due';
        task.manual_grading_feedback_on = 'grade';
        expect.snapshot(<C><Instructions {...props} /></C>).toMatchSnapshot();
    });

    it('renders the assignment instructions for feedback when published', () => {
        task.auto_grading_feedback_on = 'publish';
        task.manual_grading_feedback_on = 'publish';
        expect.snapshot(<C><Instructions {...props} /></C>).toMatchSnapshot();
    });
});
