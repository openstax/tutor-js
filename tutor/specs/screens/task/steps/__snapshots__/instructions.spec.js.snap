// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions renders the assignment instructions for feedback when due 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-height: 400px;
  border-radius: 0.25rem;
  margin: 0 auto 5rem auto;
  border: 1px solid #e4e4e4;
  border-radius: 0.25rem;
  background-color: white;
}

.c0 {
  padding: 2rem;
}

.c8 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 4rem;
  float: right;
}

.c2 {
  padding-bottom: 5rem;
}

.c3 {
  border-left: 8px solid #34bdd7;
  background-color: #daf3f8;
  font-size: 1.8rem;
  font-weight: bold;
  padding: 1.5rem 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: inherit;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 1.6rem;
  font-weight: normal;
}

.c5 {
  padding: 4rem 16rem;
}

.c5 p {
  margin: 0.5rem 0 3.2rem 0;
  line-height: 2rem;
}

.c5 ul {
  margin-bottom: 3.2rem;
  padding-left: 15px;
}

.c5 li {
  margin: 0.5rem 0;
}

.c7 {
  border-top: 1px solid #6f6f6f;
  margin: 0 16rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 .btn {
  margin: 2rem 0;
  padding: 1rem;
}

.c6 {
  font-weight: bold;
}

@media (min-width:1000px) {
  .c1 {
    max-width: 1000px;
    min-width: 750px;
  }
}

@media (min-width:1600px) {
  .c1 {
    min-width: 1000px;
  }
}

@media (max-width:999px) {
  .c0 {
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    height: calc(100vh - 60px - 50px);
    margin: 0;
  }
}

@media (max-width:999px) {
  .c3 .instructions-text {
    display: none;
  }
}

@media (max-width:600px) {
  .c3 {
    padding: 1.5rem 1rem;
  }
}

@media (max-width:600px) {
  .c5 {
    padding: 2rem 1.5rem 1rem;
  }
}

@media (min-width:601px) and (max-width:999px) {
  .c5 {
    padding: 2rem 3rem 10rem;
  }
}

@media (max-width:600px) {
  .c7 {
    position: absolute;
    bottom: 0;
  }
}

@media (max-width:600px) {
  .c7 {
    margin: 0;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 100%;
  }

  .c7 .btn {
    padding: 2rem 6rem;
  }
}

@media (min-width:601px) and (max-width:999px) {
  .c7 {
    margin: 0 3rem;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c7 .btn {
    padding: 1rem 6rem;
  }
}

<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="c0"
      >
        <div
          className="c1 c2"
          data-test-id="homework-instructions"
        >
          <h2
            className="c3 heading"
          >
            <span>
              Assignment details 
              <span
                className="instructions-text"
              >
                and instructions
              </span>
            </span>
            <span
              className="c4"
            >
              1.0
               points
            </span>
          </h2>
          <div
            className="c5"
          >
            <div
              className="c6"
            >
              Instructor notes
            </div>
            <p>
              uniform
            </p>
            <div
              className="c6"
            >
              Due date
            </div>
            <p>
              <time
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                Mon, Aug 16, 3:00pm
              </time>
            </p>
            <div
              className="c6"
            >
              Close date
            </div>
            <p>
              <time
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                Mon, Aug 23, 3:00pm
              </time>
            </p>
            <div
              className="c6"
            >
              Late work policy
            </div>
            <p>
              0%
               of points earned after the due date will be deducted for each late 
              day
              .
            </p>
            <div
              className="c6"
            >
              Feedback time
            </div>
            <p>
              <div>
                MCQs: 
                After the due date
              </div>
              <div>
                WRQs: 
                Immediately after instructor grades
              </div>
            </p>
          </div>
          <div
            className="c7"
          >
            <button
              className="c8 continue btn btn-primary btn-lg"
              data-test-id="value-prop-continue-btn"
              disabled={false}
              onClick={[MockFunction]}
              type="button"
            >
              Review
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    Ï€
  </a>
</div>
`;

exports[`Instructions renders the assignment instructions for feedback when published 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-height: 400px;
  border-radius: 0.25rem;
  margin: 0 auto 5rem auto;
  border: 1px solid #e4e4e4;
  border-radius: 0.25rem;
  background-color: white;
}

.c0 {
  padding: 2rem;
}

.c8 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 4rem;
  float: right;
}

.c2 {
  padding-bottom: 5rem;
}

.c3 {
  border-left: 8px solid #34bdd7;
  background-color: #daf3f8;
  font-size: 1.8rem;
  font-weight: bold;
  padding: 1.5rem 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: inherit;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 1.6rem;
  font-weight: normal;
}

.c5 {
  padding: 4rem 16rem;
}

.c5 p {
  margin: 0.5rem 0 3.2rem 0;
  line-height: 2rem;
}

.c5 ul {
  margin-bottom: 3.2rem;
  padding-left: 15px;
}

.c5 li {
  margin: 0.5rem 0;
}

.c7 {
  border-top: 1px solid #6f6f6f;
  margin: 0 16rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 .btn {
  margin: 2rem 0;
  padding: 1rem;
}

.c6 {
  font-weight: bold;
}

@media (min-width:1000px) {
  .c1 {
    max-width: 1000px;
    min-width: 750px;
  }
}

@media (min-width:1600px) {
  .c1 {
    min-width: 1000px;
  }
}

@media (max-width:999px) {
  .c0 {
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    height: calc(100vh - 60px - 50px);
    margin: 0;
  }
}

@media (max-width:999px) {
  .c3 .instructions-text {
    display: none;
  }
}

@media (max-width:600px) {
  .c3 {
    padding: 1.5rem 1rem;
  }
}

@media (max-width:600px) {
  .c5 {
    padding: 2rem 1.5rem 1rem;
  }
}

@media (min-width:601px) and (max-width:999px) {
  .c5 {
    padding: 2rem 3rem 10rem;
  }
}

@media (max-width:600px) {
  .c7 {
    position: absolute;
    bottom: 0;
  }
}

@media (max-width:600px) {
  .c7 {
    margin: 0;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 100%;
  }

  .c7 .btn {
    padding: 2rem 6rem;
  }
}

@media (min-width:601px) and (max-width:999px) {
  .c7 {
    margin: 0 3rem;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c7 .btn {
    padding: 1rem 6rem;
  }
}

<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="c0"
      >
        <div
          className="c1 c2"
          data-test-id="homework-instructions"
        >
          <h2
            className="c3 heading"
          >
            <span>
              Assignment details 
              <span
                className="instructions-text"
              >
                and instructions
              </span>
            </span>
            <span
              className="c4"
            >
              1.0
               points
            </span>
          </h2>
          <div
            className="c5"
          >
            <div
              className="c6"
            >
              Instructor notes
            </div>
            <p>
              tangible
            </p>
            <div
              className="c6"
            >
              Due date
            </div>
            <p>
              <time
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                Mon, Aug 16, 3:00pm
              </time>
            </p>
            <div
              className="c6"
            >
              Close date
            </div>
            <p>
              <time
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                Mon, Aug 23, 3:00pm
              </time>
            </p>
            <div
              className="c6"
            >
              Late work policy
            </div>
            <p>
              100%
               of points earned after the due date will be deducted for each late 
              assignment
              .
            </p>
            <div
              className="c6"
            >
              Feedback time
            </div>
            <p>
              <div>
                MCQs: 
                After scores are published
              </div>
              <div>
                WRQs: 
                After scores are published
              </div>
            </p>
          </div>
          <div
            className="c7"
          >
            <button
              className="c8 continue btn btn-primary btn-lg"
              data-test-id="value-prop-continue-btn"
              disabled={false}
              onClick={[MockFunction]}
              type="button"
            >
              Start
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    Ï€
  </a>
</div>
`;

exports[`Instructions renders the assignment instructions for immediate feedback 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-height: 400px;
  border-radius: 0.25rem;
  margin: 0 auto 5rem auto;
  border: 1px solid #e4e4e4;
  border-radius: 0.25rem;
  background-color: white;
}

.c0 {
  padding: 2rem;
}

.c8 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 4rem;
  float: right;
}

.c2 {
  padding-bottom: 5rem;
}

.c3 {
  border-left: 8px solid #34bdd7;
  background-color: #daf3f8;
  font-size: 1.8rem;
  font-weight: bold;
  padding: 1.5rem 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: inherit;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 1.6rem;
  font-weight: normal;
}

.c5 {
  padding: 4rem 16rem;
}

.c5 p {
  margin: 0.5rem 0 3.2rem 0;
  line-height: 2rem;
}

.c5 ul {
  margin-bottom: 3.2rem;
  padding-left: 15px;
}

.c5 li {
  margin: 0.5rem 0;
}

.c7 {
  border-top: 1px solid #6f6f6f;
  margin: 0 16rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 .btn {
  margin: 2rem 0;
  padding: 1rem;
}

.c6 {
  font-weight: bold;
}

@media (min-width:1000px) {
  .c1 {
    max-width: 1000px;
    min-width: 750px;
  }
}

@media (min-width:1600px) {
  .c1 {
    min-width: 1000px;
  }
}

@media (max-width:999px) {
  .c0 {
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    height: calc(100vh - 60px - 50px);
    margin: 0;
  }
}

@media (max-width:999px) {
  .c3 .instructions-text {
    display: none;
  }
}

@media (max-width:600px) {
  .c3 {
    padding: 1.5rem 1rem;
  }
}

@media (max-width:600px) {
  .c5 {
    padding: 2rem 1.5rem 1rem;
  }
}

@media (min-width:601px) and (max-width:999px) {
  .c5 {
    padding: 2rem 3rem 10rem;
  }
}

@media (max-width:600px) {
  .c7 {
    position: absolute;
    bottom: 0;
  }
}

@media (max-width:600px) {
  .c7 {
    margin: 0;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 100%;
  }

  .c7 .btn {
    padding: 2rem 6rem;
  }
}

@media (min-width:601px) and (max-width:999px) {
  .c7 {
    margin: 0 3rem;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c7 .btn {
    padding: 1rem 6rem;
  }
}

<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="c0"
      >
        <div
          className="c1 c2"
          data-test-id="homework-instructions"
        >
          <h2
            className="c3 heading"
          >
            <span>
              Assignment details 
              <span
                className="instructions-text"
              >
                and instructions
              </span>
            </span>
            <span
              className="c4"
            >
              1.0
               points
            </span>
          </h2>
          <div
            className="c5"
          >
            <div
              className="c6"
            >
              Instructor notes
            </div>
            <p>
              6th generation
            </p>
            <div
              className="c6"
            >
              Due date
            </div>
            <p>
              <time
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                Mon, Aug 16, 3:00pm
              </time>
            </p>
            <div
              className="c6"
            >
              Close date
            </div>
            <p>
              <time
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                Mon, Aug 23, 3:00pm
              </time>
            </p>
            <div
              className="c6"
            >
              Late work policy
            </div>
            <p>
              100%
               of points earned after the due date will be deducted for each late 
              assignment
              .
            </p>
            <div
              className="c6"
            >
              Feedback time
            </div>
            <p>
              <div>
                MCQs: 
                Immediately after answer is submitted
              </div>
              <div>
                WRQs: 
                Immediately after instructor grades
              </div>
            </p>
          </div>
          <div
            className="c7"
          >
            <button
              className="c8 continue btn btn-primary btn-lg"
              data-test-id="value-prop-continue-btn"
              disabled={false}
              onClick={[MockFunction]}
              type="button"
            >
              Start
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    Ï€
  </a>
</div>
`;
